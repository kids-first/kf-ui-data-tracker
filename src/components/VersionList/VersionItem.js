import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {Mutation} from 'react-apollo';
import {FILE_DOWNLOAD_URL} from '../../state/mutations';
import {Avatar, Icon, GridContainer} from 'kf-uikit';
import TimeAgo from 'react-timeago';

import {
  fileTypeDetail,
  versionState,
  formatFileSize,
  downloadFile,
} from '../../common/fileUtils';
import SvgIcon from '../Icon/Icon';
/**
 * Displays single version item from the list
 */
const VersionItem = ({
  className,
  studyId,
  fileId,
  fileType,
  versionNode,
  index,
  onNameClick,
}) => {
  let versionItemClass = classes('FileVersionList--Element', className);
  const stateColor = versionNode.state
    ? versionState[versionNode.state].color
    : 'bg-lightGrey';
  const iconClass = classes(
    'FileVersionElement--Icon',
    'row-3',
    'cell-1',
    'lg:row-1',
    'lg:cell-1',
    stateColor,
  );
  return (
    <Mutation mutation={FILE_DOWNLOAD_URL}>
      {downloadFileMutation => (
        <li key={versionNode.kfId} className={versionItemClass}>
          <GridContainer collapsed="rows" className="p-4">
            <div className={iconClass}>
              <SvgIcon
                kind={fileTypeDetail[fileType].icon}
                width="14"
                height="14"
              />
            </div>
            {index === 0 && (
              <div className="FileVersionElement--Tag row-3 cell-2 lg:row-2 lg:cell-1">
                Latest
              </div>
            )}
            <div className="flex row-1 cell-12 lg:cell-7">
              <p
                className="FileVersionElement--button m-0 p-0 text-lg"
                onClick={e => onNameClick(versionNode, index)}
              >
                {versionNode.fileName}
              </p>
            </div>
            <div className="flex row-3 cell-9 lg:row-1 lg:cell-4 lg:justify-end align-center">
              <div className="flex align-center">
                <Avatar
                  className="inline-block mr-8"
                  size={20}
                  imgUrl={
                    versionNode.creator
                      ? versionNode.creator.picture
                      : 'https://www.w3schools.com/css/img_avatar.png'
                  }
                  userName={versionNode.creator && versionNode.creator.username}
                  userEmail={versionNode.creator && versionNode.creator.email}
                />
                <TimeAgo
                  date={versionNode.createdAt}
                  live={false}
                  className="FileVersionList--text  FileVersionList--Element-meta"
                />
              </div>
              <p className=" FileVersionList--text FileVersionList--Element-meta row-1 cell-2">
                {formatFileSize(versionNode.size, true) || 'Size Unknown'}
              </p>
              <button
                onClick={e =>
                  downloadFile(
                    studyId,
                    fileId,
                    versionNode.kfId,
                    downloadFileMutation,
                  )
                }
                className="FileVersionList--text row-1 cell-1"
              >
                <Icon kind="download" width={12} height={12} />
              </button>
            </div>

            <p className="FileVersionList--Element-desc FileVersionList--text row-2 cell-12 lg:cell-2-11 text-xs">
              {versionNode.description || 'No version summary available'}
            </p>
          </GridContainer>
        </li>
      )}
    </Mutation>
  );
};

VersionItem.propTypes = {
  /** Any additional classes to be applied to the study list*/
  className: PropTypes.string,
  /** Study kfId*/
  studyId: PropTypes.string,
  /** File kfId*/
  fileId: PropTypes.string,
  /** File type*/
  fileType: PropTypes.string,
  /** Version object*/
  versionNode: PropTypes.object.isRequired,
  /** Index of version item (sort by createdAt)*/
  index: PropTypes.number,
};

VersionItem.defaultProps = {
  className: null,
  studyId: null,
  fileId: null,
  fileType: null,
  fileNode: {},
  index: null,
};

export default VersionItem;
