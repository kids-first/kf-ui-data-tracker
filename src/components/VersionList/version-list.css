.FileVersionContainer {
  border-radius: 5px;
  border-color: #cccccc;
  @extend .border;
}

.FileVersionList {
  @extend .list-reset, .min-w-full, .m-0;
}

.FileVersionList--Button {
  @extend .inline-block,
    .bg-white,
    .p-4,
    .px-12,
    .border,
    .border-greyBorder,
    .rounded-lg,
    .font-bold,
    .font-title,
    .text-xs,
    .text-tealA11y,
    .uppercase,
    .flex,
    .leading-tight,
    .align-middle,
    .whitespace-no-wrap,
    .h-full;
}

.FileVersionList--Button:hover {
  @extend .bg-lightGrey, .text-tealHover;
}

.FileVersionList--Element {
  @extend .border-t, .border-mediumGrey. .list-reset, .min-w-full;
}

.FileVersionList--hidden {
  @extend .invisible;
}

.FileVersionList--Element:hover .FileVersionList--hidden {
  @extend .visible;
}

.FileVersionList--Element:hover {
  @extend .bg-lightGrey;
}

.FileVersionList--text {
  @extend .my-0,
    .mr-8,
    .text-sm,
    .text-grey,
    .whitespace-no-wrap,
    .overflow-hidden;
  text-overflow: ellipsis;
}

.FileVersionElement--button {
  @extend .text-sm, .font-black, .text-left, .text-blue;
}

.FileVersionElement--Icon {
  margin-left: auto;
  margin-right: auto;
  @extend .rounded-full, .w-6, .h-6, .flex, .justify-center, .items-center;
}

.FileVersionElement--Tag {
  border-radius: 3px;
  background-color: #d9f5ff;
  @extend .w-12,
    .mt-4,
    .text-xs,
    .text-center,
    .border,
    .border-teal,
    .text-teal;
}
