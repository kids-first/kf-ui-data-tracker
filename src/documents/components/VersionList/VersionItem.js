import React from 'react';
import PropTypes from 'prop-types';
import {useMutation} from '@apollo/client';
import {FILE_DOWNLOAD_URL} from '../../mutations';
import AvatarTimeAgo from '../../../components/AvatarTimeAgo/AvatarTimeAgo';
import {formatFileSize, downloadFile, lengthLimit} from '../../utilities';
import {Label, Icon, Table} from 'semantic-ui-react';
/**
 * Displays single version item from the list
 */
const VersionItem = ({
  studyId,
  fileId,
  fileType,
  versionNode,
  index,
  onNameClick,
}) => {
  const [downloadFileMutation] = useMutation(FILE_DOWNLOAD_URL);
  const size = versionNode.size
    ? formatFileSize(versionNode.size, true)
    : 'Size Unknown';
  return (
    <Table.Row
      onClick={e => onNameClick(versionNode, index)}
      className="version--item"
    >
      <Table.Cell>
        <p title={versionNode.fileName}>
          <b>{lengthLimit(versionNode.fileName, 70)}</b>
        </p>
        <p title={versionNode.description} data-testid="version-item">
          <small>
            {lengthLimit(versionNode.description, 110)}
            {versionNode.description.length === 0 &&
              'No version summary available'}
          </small>
        </p>
      </Table.Cell>
      <Table.Cell textAlign="right" verticalAlign="top" collapsing>
        <AvatarTimeAgo
          size="mini"
          creator={versionNode.creator}
          createdAt={versionNode.createdAt}
        />
        <Label
          basic
          size="mini"
          as="button"
          onClick={e => {
            e.stopPropagation();
            downloadFile(
              studyId,
              fileId,
              versionNode.kfId,
              downloadFileMutation,
            );
          }}
        >
          <Icon name="download" />
          {size}
        </Label>
      </Table.Cell>
    </Table.Row>
  );
};

VersionItem.propTypes = {
  /** Study kfId*/
  studyId: PropTypes.string,
  /** File kfId*/
  fileId: PropTypes.string,
  /** File type*/
  fileType: PropTypes.string,
  /** Version object*/
  versionNode: PropTypes.object.isRequired,
  /** Index of version item (sort by createdAt)*/
  index: PropTypes.number,
};

VersionItem.defaultProps = {
  studyId: null,
  fileId: null,
  fileType: null,
  fileNode: {},
  index: null,
};

export default VersionItem;
