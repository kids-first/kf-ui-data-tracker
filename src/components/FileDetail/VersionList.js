import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {dateCompare} from '../../common/fileUtils';
import VersionItem from './VersionItem';
import {CREATE_FILE} from '../../state/mutations';
import {GET_FILE_BY_ID} from '../../state/queries';
import {Query, Mutation} from 'react-apollo';
import SvgIcon from '../Icon/Icon';
/**
 * Displays ordered versions of one file. (Latest first)
 */
const VersionList = ({className, studyId, fileNode, onUploadClick}) => {
  let versionListClass = classes('FileVersionList', className);

  return (
    <Query query={GET_FILE_BY_ID} variables={{kfId: fileNode.kfId}}>
      {({loading, error, data}) => {
        if (error) return `Error!: ${error}`;
        return (
          <Mutation
            mutation={CREATE_FILE}
            refetchQueries={res => [
              {
                query: GET_FILE_BY_ID,
                variables: {kfId: fileNode.kfId},
              },
            ]}
          >
            {(createFile, {data, error}) => (
              <div className="lg:cell-10 md:cell-9 cell-12">
                <div className="bg-lightGrey p-8 flex justify-between">
                  <p className="FileInfo--Title">
                    Versions
                    <span className="font-light ml-4">
                      ({fileNode.versions.edges.length})
                    </span>
                  </p>

                  <label className="FileVersionList--Button">
                    <SvgIcon
                      kind="Upload"
                      width="14"
                      height="14"
                      className="mr-4"
                    />
                    UPLOAD VERSION
                    <input
                      className="hidden"
                      type="file"
                      onChange={e => {
                        e.preventDefault();
                        createFile({
                          variables: {
                            file: e.target.files[0],
                            studyId,
                            fileId: fileNode.kfId,
                          },
                        })
                          .then(resp => {})
                          .catch(err => {
                            console.log(err);
                            alert('Failed to upload file.');
                          });
                      }}
                    />
                  </label>
                </div>
                <ul className={versionListClass}>
                  {fileNode.versions.edges.length ? (
                    fileNode.versions.edges
                      .sort(dateCompare)
                      .map(({node}, index) => (
                        <VersionItem
                          key={node.kfId}
                          studyId={studyId}
                          fileId={fileNode.kfId}
                          fileType={fileNode.fileType}
                          fileName={fileNode.name}
                          versionNode={node}
                          index={index}
                        />
                      ))
                  ) : (
                    <h3>You don't have any versions of this file yet.</h3>
                  )}
                </ul>
              </div>
            )}
          </Mutation>
        );
      }}
    </Query>
  );
};

VersionList.propTypes = {
  /** Any additional classes to be applied to the version list*/
  className: PropTypes.string,
  /** Study kfId*/
  studyId: PropTypes.string,
  /** File object*/
  fileNode: PropTypes.object.isRequired,
  /** Action when click on the upload button*/
  onUploadClick: PropTypes.func,
};

VersionList.defaultProps = {
  className: null,
  studyId: null,
  fileNode: {},
};

export default VersionList;
