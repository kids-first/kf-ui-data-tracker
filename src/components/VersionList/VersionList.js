import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {dateCompare} from '../../common/fileUtils';
import VersionItem from './VersionItem';
import SvgIcon from '../Icon/Icon';
/**
 * Displays ordered versions of one file. (Latest first)
 */
const VersionList = ({className, studyId, fileNode, onUploadClick}) => {
  let versionListClass = classes('FileVersionList', className);

  return (
    <div className="FileVersionContainer lg:cell-10 md:cell-9 cell-12">
      <div className="p-8 flex justify-between">
        <p className="FileInfo--Title">
          Document Versions
          <span className="font-light ml-4">
            ({fileNode.versions.edges.length})
          </span>
        </p>
        <button className="FileVersionList--Button" onClick={onUploadClick}>
          <SvgIcon kind="Upload" height="15" width="15" className="mr-4" />
          UPLOAD VERSION
        </button>
      </div>
      <ul className={versionListClass}>
        {fileNode.versions.edges.length ? (
          fileNode.versions.edges
            .sort(dateCompare)
            .map(({node}, index) => (
              <VersionItem
                key={index}
                studyId={studyId}
                fileId={fileNode.kfId}
                fileType={fileNode.fileType}
                versionNode={node}
                index={index}
              />
            ))
        ) : (
          <h3>You don't have any versions of this file yet.</h3>
        )}
      </ul>
    </div>
  );
};

VersionList.propTypes = {
  /** Any additional classes to be applied to the version list*/
  className: PropTypes.string,
  /** Study kfId*/
  studyId: PropTypes.string,
  /** File object*/
  fileNode: PropTypes.object.isRequired,
  /** Action when click on the upload button*/
  onUploadClick: PropTypes.func,
};

VersionList.defaultProps = {
  className: null,
  studyId: null,
  fileNode: {},
};

export default VersionList;
