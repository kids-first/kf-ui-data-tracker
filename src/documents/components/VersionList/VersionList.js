import React from 'react';
import PropTypes from 'prop-types';
import {dateCompare} from '../../utilities';
import VersionItem from './VersionItem';
import {Button, Table} from 'semantic-ui-react';
/**
 * Displays ordered versions of one file. (Latest first)
 */
const VersionList = ({
  studyId,
  fileNode,
  onUploadClick,
  onNameClick,
  allowUpload,
}) => {
  return (
    <Table compact="very" selectable>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell
            singleLine
            colSpan="3"
            textAlign="center"
            verticalAlign="bottom"
          >
            Document Versions ({fileNode.versions.edges.length})
            {allowUpload && (
              <Button
                compact
                primary
                floated="right"
                onClick={onUploadClick}
                labelPosition="left"
                size="mini"
                icon="cloud upload"
                content="UPLOAD VERSION"
              />
            )}
          </Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {fileNode.versions.edges.length ? (
          fileNode.versions.edges
            .sort(dateCompare)
            .map(({node}, index) => (
              <VersionItem
                key={index}
                studyId={studyId}
                fileId={fileNode.kfId}
                fileType={fileNode.fileType}
                versionNode={node}
                index={index}
                onNameClick={onNameClick}
              />
            ))
        ) : (
          <Table.Row>
            <Table.Cell colSpan="3" textAlign="center">
              You don't have any versions of this file yet.
            </Table.Cell>
          </Table.Row>
        )}
      </Table.Body>
    </Table>
  );
};

VersionList.propTypes = {
  /** Study kfId*/
  studyId: PropTypes.string,
  /** File object*/
  fileNode: PropTypes.object.isRequired,
  /** Action when click on the upload button*/
  onUploadClick: PropTypes.func,
};

VersionList.defaultProps = {
  studyId: null,
  fileNode: {},
};

export default VersionList;
