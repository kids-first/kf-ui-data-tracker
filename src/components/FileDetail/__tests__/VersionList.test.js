import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {MockedProvider} from 'react-apollo/test-utils';
import {render} from 'react-testing-library';
import VersionList from '../VersionList';
import fileByKfId from './fileByKfId';

it('renders with more than one versions', () => {
  const fileNodeMultipleVersion = fileByKfId[0].data.fileByKfId;

  const tree = render(
    <MemoryRouter>
      <MockedProvider>
        <VersionList
          fileNode={fileNodeMultipleVersion}
          studyId={'SD_00000000'}
        />
      </MockedProvider>
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();

  const lis = tree.container.querySelectorAll('.FileVersionList--Element');
  expect(lis.length).toBe(4);
});

it('renders with only one version', () => {
  const fileNodeOneVersion = fileByKfId[1].data.fileByKfId;

  const tree = render(
    <MemoryRouter>
      <MockedProvider>
        <VersionList fileNode={fileNodeOneVersion} studyId={'SD_00000000'} />
      </MockedProvider>
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();

  const lis = tree.container.querySelectorAll('.FileVersionList--Element');
  expect(lis.length).toBe(1);
});
