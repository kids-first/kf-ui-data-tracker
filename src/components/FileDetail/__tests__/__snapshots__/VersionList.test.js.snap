// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with more than one versions 1`] = `
<div>
  <div
    class="lg:cell-10 md:cell-9 cell-12"
  >
    <div
      class="bg-lightGrey p-8 flex justify-between"
    >
      <p
        class="FileInfo--Title"
      >
        Versions
        <span
          class="font-light ml-4"
        >
          (
          4
          )
        </span>
      </p>
      <label
        class="FileVersionList--Button"
      >
        <svg
          class="mr-4"
          height="14"
          viewBox="0 0 640 512"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M537.6 226.6c4.1-10.7 6.4-22.4 6.4-34.6 0-53-43-96-96-96-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32c-88.4 0-160 71.6-160 160 0 2.7.1 5.4.2 8.1C40.2 219.8 0 273.2 0 336c0 79.5 64.5 144 144 144h368c70.7 0 128-57.3 128-128 0-61.9-44-113.6-102.4-125.4zM393.4 288H328v112c0 8.8-7.2 16-16 16h-48c-8.8 0-16-7.2-16-16V288h-65.4c-14.3 0-21.4-17.2-11.3-27.3l105.4-105.4c6.2-6.2 16.4-6.2 22.6 0l105.4 105.4c10.1 10.1 2.9 27.3-11.3 27.3z"
          />
        </svg>
        UPLOAD VERSION
        <input
          class="hidden"
          type="file"
        />
      </label>
    </div>
    <ul
      class="FileVersionList"
    >
      <li
        class="FileVersionList--Element"
      >
        <svg
          class="mx-16"
          height="12"
          viewBox="0 0 94 94"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <title>
            icon-error
          </title>
          <g
            data-name="Layer 1"
            id="Layer_1"
          >
            <circle
              cx="47"
              cy="47"
              fill="#fff"
              r="43"
            />
            <path
              d="M47,94A47,47,0,1,1,94,47,47,47,0,0,1,47,94ZM47,8A39,39,0,1,0,86,47,39,39,0,0,0,47,8Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
            <path
              d="M59.83,64.07a4.25,4.25,0,0,1-3-1.26L47,53.05,37.2,62.83a4.32,4.32,0,0,1-6.05,0,4.26,4.26,0,0,1,0-6L41,47l-9.79-9.8a4.3,4.3,0,0,1,0-6.05,4.38,4.38,0,0,1,6.07,0L47.06,41l9.7-9.65a4.3,4.3,0,0,1,6,0,4.24,4.24,0,0,1,0,6.05L53.11,47l9.74,9.74a4.3,4.3,0,0,1,0,6.06,4.25,4.25,0,0,1-3,1.26Zm-1-30.75h0Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
          </g>
        </svg>
        <button
          class="FileVersionElement--button w-24"
        >
          Latest
        </button>
        <span
          class="FileVersionElement--button FileVersionList--hidden w-12"
        >
          <button>
            <svg
              height="10"
              id="mock-icon"
              kind="copy"
              width="10"
            >
              <title>
                icon-
                copy
              </title>
            </svg>
          </button>
        </span>
        <div
          class="Avatar inline-block mr-8 hidden"
          style="height: 20px; width: 20px;"
        >
          <img
            alt="null avatar"
            class="Avatar-img"
            src="https://www.w3schools.com/css/img_avatar.png"
          />
        </div>
        <time
          class="FileVersionList--text w-32"
          datetime="2019-05-16T15:13:02.090Z"
          title="2019-05-16T15:13:02.090275+00:00"
        >
          3 weeks from now
        </time>
        <p
          class="FileVersionList--text w-16 hidden"
        >
          .jpg
        </p>
        <p
          class="FileVersionList--text w-24"
        >
          61.0 kB
        </p>
        <div
          class="flex"
        >
          <div
            class="FileVersionList--icon"
          >
            <svg
              height="12"
              id="mock-icon"
              kind="settings"
              width="12"
            >
              <title>
                icon-
                settings
              </title>
            </svg>
          </div>
          <p
            class="FileVersionList--text my-0 w-48"
          >
            Other
          </p>
        </div>
        <button>
          <svg
            class="FileVersionList--hidden w-12"
            height="12"
            id="mock-icon"
            kind="download"
            width="12"
          >
            <title>
              icon-
              download
            </title>
          </svg>
        </button>
      </li>
      <li
        class="FileVersionList--Element"
      >
        <svg
          class="mx-16"
          height="12"
          viewBox="0 0 94 94"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <title>
            icon-error
          </title>
          <g
            data-name="Layer 1"
            id="Layer_1"
          >
            <circle
              cx="47"
              cy="47"
              fill="#fff"
              r="43"
            />
            <path
              d="M47,94A47,47,0,1,1,94,47,47,47,0,0,1,47,94ZM47,8A39,39,0,1,0,86,47,39,39,0,0,0,47,8Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
            <path
              d="M59.83,64.07a4.25,4.25,0,0,1-3-1.26L47,53.05,37.2,62.83a4.32,4.32,0,0,1-6.05,0,4.26,4.26,0,0,1,0-6L41,47l-9.79-9.8a4.3,4.3,0,0,1,0-6.05,4.38,4.38,0,0,1,6.07,0L47.06,41l9.7-9.65a4.3,4.3,0,0,1,6,0,4.24,4.24,0,0,1,0,6.05L53.11,47l9.74,9.74a4.3,4.3,0,0,1,0,6.06,4.25,4.25,0,0,1-3,1.26Zm-1-30.75h0Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
          </g>
        </svg>
        <button
          class="FileVersionElement--button w-24"
        >
          FV_NVF2M1V9
        </button>
        <span
          class="FileVersionElement--button FileVersionList--hidden w-12"
        >
          <button>
            <svg
              height="10"
              id="mock-icon"
              kind="copy"
              width="10"
            >
              <title>
                icon-
                copy
              </title>
            </svg>
          </button>
        </span>
        <div
          class="Avatar inline-block mr-8 hidden"
          style="height: 20px; width: 20px;"
        >
          <img
            alt="null avatar"
            class="Avatar-img"
            src="https://www.w3schools.com/css/img_avatar.png"
          />
        </div>
        <time
          class="FileVersionList--text w-32"
          datetime="2019-05-16T15:12:58.888Z"
          title="2019-05-16T15:12:58.888683+00:00"
        >
          3 weeks from now
        </time>
        <p
          class="FileVersionList--text w-16 hidden"
        >
          .jpg
        </p>
        <p
          class="FileVersionList--text w-24"
        >
          75.5 kB
        </p>
        <div
          class="flex"
        >
          <div
            class="FileVersionList--icon"
          >
            <svg
              height="12"
              id="mock-icon"
              kind="settings"
              width="12"
            >
              <title>
                icon-
                settings
              </title>
            </svg>
          </div>
          <p
            class="FileVersionList--text my-0 w-48"
          >
            Other
          </p>
        </div>
        <button>
          <svg
            class="FileVersionList--hidden w-12"
            height="12"
            id="mock-icon"
            kind="download"
            width="12"
          >
            <title>
              icon-
              download
            </title>
          </svg>
        </button>
      </li>
      <li
        class="FileVersionList--Element"
      >
        <svg
          class="mx-16"
          height="12"
          viewBox="0 0 94 94"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <title>
            icon-error
          </title>
          <g
            data-name="Layer 1"
            id="Layer_1"
          >
            <circle
              cx="47"
              cy="47"
              fill="#fff"
              r="43"
            />
            <path
              d="M47,94A47,47,0,1,1,94,47,47,47,0,0,1,47,94ZM47,8A39,39,0,1,0,86,47,39,39,0,0,0,47,8Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
            <path
              d="M59.83,64.07a4.25,4.25,0,0,1-3-1.26L47,53.05,37.2,62.83a4.32,4.32,0,0,1-6.05,0,4.26,4.26,0,0,1,0-6L41,47l-9.79-9.8a4.3,4.3,0,0,1,0-6.05,4.38,4.38,0,0,1,6.07,0L47.06,41l9.7-9.65a4.3,4.3,0,0,1,6,0,4.24,4.24,0,0,1,0,6.05L53.11,47l9.74,9.74a4.3,4.3,0,0,1,0,6.06,4.25,4.25,0,0,1-3,1.26Zm-1-30.75h0Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
          </g>
        </svg>
        <button
          class="FileVersionElement--button w-24"
        >
          FV_KXMGAKK4
        </button>
        <span
          class="FileVersionElement--button FileVersionList--hidden w-12"
        >
          <button>
            <svg
              height="10"
              id="mock-icon"
              kind="copy"
              width="10"
            >
              <title>
                icon-
                copy
              </title>
            </svg>
          </button>
        </span>
        <div
          class="Avatar inline-block mr-8 hidden"
          style="height: 20px; width: 20px;"
        >
          <img
            alt="null avatar"
            class="Avatar-img"
            src="https://www.w3schools.com/css/img_avatar.png"
          />
        </div>
        <time
          class="FileVersionList--text w-32"
          datetime="2019-05-16T15:12:55.598Z"
          title="2019-05-16T15:12:55.598416+00:00"
        >
          3 weeks from now
        </time>
        <p
          class="FileVersionList--text w-16 hidden"
        >
          .jpg
        </p>
        <p
          class="FileVersionList--text w-24"
        >
          63.0 kB
        </p>
        <div
          class="flex"
        >
          <div
            class="FileVersionList--icon"
          >
            <svg
              height="12"
              id="mock-icon"
              kind="settings"
              width="12"
            >
              <title>
                icon-
                settings
              </title>
            </svg>
          </div>
          <p
            class="FileVersionList--text my-0 w-48"
          >
            Other
          </p>
        </div>
        <button>
          <svg
            class="FileVersionList--hidden w-12"
            height="12"
            id="mock-icon"
            kind="download"
            width="12"
          >
            <title>
              icon-
              download
            </title>
          </svg>
        </button>
      </li>
      <li
        class="FileVersionList--Element"
      >
        <svg
          class="mx-16"
          height="12"
          viewBox="0 0 94 94"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <title>
            icon-error
          </title>
          <g
            data-name="Layer 1"
            id="Layer_1"
          >
            <circle
              cx="47"
              cy="47"
              fill="#fff"
              r="43"
            />
            <path
              d="M47,94A47,47,0,1,1,94,47,47,47,0,0,1,47,94ZM47,8A39,39,0,1,0,86,47,39,39,0,0,0,47,8Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
            <path
              d="M59.83,64.07a4.25,4.25,0,0,1-3-1.26L47,53.05,37.2,62.83a4.32,4.32,0,0,1-6.05,0,4.26,4.26,0,0,1,0-6L41,47l-9.79-9.8a4.3,4.3,0,0,1,0-6.05,4.38,4.38,0,0,1,6.07,0L47.06,41l9.7-9.65a4.3,4.3,0,0,1,6,0,4.24,4.24,0,0,1,0,6.05L53.11,47l9.74,9.74a4.3,4.3,0,0,1,0,6.06,4.25,4.25,0,0,1-3,1.26Zm-1-30.75h0Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
          </g>
        </svg>
        <button
          class="FileVersionElement--button w-24"
        >
          FV_MGG87VHC
        </button>
        <span
          class="FileVersionElement--button FileVersionList--hidden w-12"
        >
          <button>
            <svg
              height="10"
              id="mock-icon"
              kind="copy"
              width="10"
            >
              <title>
                icon-
                copy
              </title>
            </svg>
          </button>
        </span>
        <div
          class="Avatar inline-block mr-8 hidden"
          style="height: 20px; width: 20px;"
        >
          <img
            alt="null avatar"
            class="Avatar-img"
            src="https://www.w3schools.com/css/img_avatar.png"
          />
        </div>
        <time
          class="FileVersionList--text w-32"
          datetime="2019-05-16T14:33:58.284Z"
          title="2019-05-16T14:33:58.284296+00:00"
        >
          3 weeks from now
        </time>
        <p
          class="FileVersionList--text w-16 hidden"
        >
          .jpg
        </p>
        <p
          class="FileVersionList--text w-24"
        >
          35.6 kB
        </p>
        <div
          class="flex"
        >
          <div
            class="FileVersionList--icon"
          >
            <svg
              height="12"
              id="mock-icon"
              kind="settings"
              width="12"
            >
              <title>
                icon-
                settings
              </title>
            </svg>
          </div>
          <p
            class="FileVersionList--text my-0 w-48"
          >
            Other
          </p>
        </div>
        <button>
          <svg
            class="FileVersionList--hidden w-12"
            height="12"
            id="mock-icon"
            kind="download"
            width="12"
          >
            <title>
              icon-
              download
            </title>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders with only one version 1`] = `
<div>
  <div
    class="lg:cell-10 md:cell-9 cell-12"
  >
    <div
      class="bg-lightGrey p-8 flex justify-between"
    >
      <p
        class="FileInfo--Title"
      >
        Versions
        <span
          class="font-light ml-4"
        >
          (
          1
          )
        </span>
      </p>
      <label
        class="FileVersionList--Button"
      >
        <svg
          class="mr-4"
          height="14"
          viewBox="0 0 640 512"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M537.6 226.6c4.1-10.7 6.4-22.4 6.4-34.6 0-53-43-96-96-96-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32c-88.4 0-160 71.6-160 160 0 2.7.1 5.4.2 8.1C40.2 219.8 0 273.2 0 336c0 79.5 64.5 144 144 144h368c70.7 0 128-57.3 128-128 0-61.9-44-113.6-102.4-125.4zM393.4 288H328v112c0 8.8-7.2 16-16 16h-48c-8.8 0-16-7.2-16-16V288h-65.4c-14.3 0-21.4-17.2-11.3-27.3l105.4-105.4c6.2-6.2 16.4-6.2 22.6 0l105.4 105.4c10.1 10.1 2.9 27.3-11.3 27.3z"
          />
        </svg>
        UPLOAD VERSION
        <input
          class="hidden"
          type="file"
        />
      </label>
    </div>
    <ul
      class="FileVersionList"
    >
      <li
        class="FileVersionList--Element"
      >
        <svg
          class="mx-16"
          height="12"
          viewBox="0 0 94 94"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <title>
            icon-error
          </title>
          <g
            data-name="Layer 1"
            id="Layer_1"
          >
            <circle
              cx="47"
              cy="47"
              fill="#fff"
              r="43"
            />
            <path
              d="M47,94A47,47,0,1,1,94,47,47,47,0,0,1,47,94ZM47,8A39,39,0,1,0,86,47,39,39,0,0,0,47,8Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
            <path
              d="M59.83,64.07a4.25,4.25,0,0,1-3-1.26L47,53.05,37.2,62.83a4.32,4.32,0,0,1-6.05,0,4.26,4.26,0,0,1,0-6L41,47l-9.79-9.8a4.3,4.3,0,0,1,0-6.05,4.38,4.38,0,0,1,6.07,0L47.06,41l9.7-9.65a4.3,4.3,0,0,1,6,0,4.24,4.24,0,0,1,0,6.05L53.11,47l9.74,9.74a4.3,4.3,0,0,1,0,6.06,4.25,4.25,0,0,1-3,1.26Zm-1-30.75h0Z"
              fill="#e45562"
              transform="translate(0 0)"
            />
          </g>
        </svg>
        <button
          class="FileVersionElement--button w-24"
        >
          Latest
        </button>
        <span
          class="FileVersionElement--button FileVersionList--hidden w-12"
        >
          <button>
            <svg
              height="10"
              id="mock-icon"
              kind="copy"
              width="10"
            >
              <title>
                icon-
                copy
              </title>
            </svg>
          </button>
        </span>
        <div
          class="Avatar inline-block mr-8 hidden"
          style="height: 20px; width: 20px;"
        >
          <img
            alt="null avatar"
            class="Avatar-img"
            src="https://www.w3schools.com/css/img_avatar.png"
          />
        </div>
        <time
          class="FileVersionList--text w-32"
          datetime="2019-05-16T15:13:50.215Z"
          title="2019-05-16T15:13:50.215303+00:00"
        >
          3 weeks from now
        </time>
        <p
          class="FileVersionList--text w-16 hidden"
        >
          .jpg
        </p>
        <p
          class="FileVersionList--text w-24"
        >
          63.0 kB
        </p>
        <div
          class="flex"
        >
          <div
            class="FileVersionList--icon"
          >
            <svg
              height="12"
              id="mock-icon"
              kind="settings"
              width="12"
            >
              <title>
                icon-
                settings
              </title>
            </svg>
          </div>
          <p
            class="FileVersionList--text my-0 w-48"
          >
            Other
          </p>
        </div>
        <button>
          <svg
            class="FileVersionList--hidden w-12"
            height="12"
            id="mock-icon"
            kind="download"
            width="12"
          >
            <title>
              icon-
              download
            </title>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
